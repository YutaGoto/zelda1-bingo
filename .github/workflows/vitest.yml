name: vitest

on:
  - push

jobs:
  vitest:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout your repository using git
        uses: actions/checkout@v4
      - name: Setup pnpm
        uses: pnpm/action-setup@v2
        with:
          version: 8
      - name: Setup Node.js
        uses: actions/setup-node@v4
        with:
          cache: "pnpm"
          node-version: 20
      - name: Install dependencies
        run: pnpm install
      - name: Run tests
        run: pnpm test
