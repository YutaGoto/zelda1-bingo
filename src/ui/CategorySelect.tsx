import { Box, Grid, Link, Text } from "@chakra-ui/react";
import { t } from "i18next";

interface CategorySelectProps {
  mode: "score" | "bingo" | "single";
  category: "firstQuest" | "secondQuest" | "swordless";
  lang: string | undefined;
  mt?: number;
}

export const CategorySelect = ({
  mode,
  category,
  lang,
  mt = 0,
}: CategorySelectProps) => {
  const modePath = mode === "bingo" ? "" : `/${mode}`;
  return (
    <Box mt={mt}>
      <Text fontSize="lg">{t("otherCategories")}</Text>
      <Grid
        templateColumns={{ lg: "repeat(2, 1fr)", md: "repeat(4, 1fr)" }}
        gap={2}
        mt={2}
      >
        <Link
          href={`${modePath}/firstQuest/${lang}`}
          variant={category === "firstQuest" ? "plain" : "underline"}
          colorScheme="blue"
        >
          1st Quest
        </Link>
        <Link
          href={`${modePath}/secondQuest/${lang}`}
          colorScheme="orange"
          variant={category === "secondQuest" ? "plain" : "underline"}
        >
          2nd Quest
        </Link>
        <Link
          href={`${modePath}/swordless/${lang}`}
          colorScheme="pink"
          variant={category === "swordless" ? "plain" : "underline"}
        >
          Swordless
        </Link>
      </Grid>
    </Box>
  );
};
