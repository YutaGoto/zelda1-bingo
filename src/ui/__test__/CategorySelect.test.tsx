import { render, screen } from "@testing-library/react";
import { describe, expect, test } from "vitest";

import { CategorySelect } from "../CategorySelect";

describe("Category Select languages", () => {
  test("title is English", () => {
    render(<CategorySelect mode="bingo" category="firstQuest" lang="en" />);
    expect(screen.getByText("Other Categories")).toBeTruthy();
  });
});

describe("active category", () => {
  test("1st quest", () => {
    const { asFragment } = render(
      <CategorySelect mode="score" category="firstQuest" lang="en" />,
    );

    expect(screen.getByText("1st Quest")).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });

  test("2nd quest", () => {
    const { asFragment } = render(
      <CategorySelect mode="score" category="secondQuest" lang="en" />,
    );

    expect(screen.getByText("2nd Quest")).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });

  test("swordless", () => {
    const { asFragment } = render(
      <CategorySelect mode="bingo" category="swordless" lang="en" />,
    );

    expect(screen.getByText("Swordless")).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });
});
