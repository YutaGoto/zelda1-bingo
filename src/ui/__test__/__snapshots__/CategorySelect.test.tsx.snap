// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`active category > 1st quest 1`] = `
<DocumentFragment>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
  </script>
  <div
    class="css-8f2oos"
  >
    <p
      class="css-uok8h6"
    >
      Other Categories
    </p>
    <div
      class="css-12atwa6"
    >
      <a
        class="chakra-link css-trbbt2"
        href="/score/firstQuest/en"
      >
        1st Quest
      </a>
      <a
        class="chakra-link css-1ejw2rc"
        href="/score/secondQuest/en"
      >
        2nd Quest
      </a>
      <a
        class="chakra-link css-1erwuv3"
        href="/score/swordless/en"
      >
        Swordless
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`active category > 2nd quest 1`] = `
<DocumentFragment>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
  </script>
  <div
    class="css-8f2oos"
  >
    <p
      class="css-uok8h6"
    >
      Other Categories
    </p>
    <div
      class="css-12atwa6"
    >
      <a
        class="chakra-link css-9dfuuy"
        href="/score/firstQuest/en"
      >
        1st Quest
      </a>
      <a
        class="chakra-link css-cbo3vp"
        href="/score/secondQuest/en"
      >
        2nd Quest
      </a>
      <a
        class="chakra-link css-1erwuv3"
        href="/score/swordless/en"
      >
        Swordless
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`active category > swordless 1`] = `
<DocumentFragment>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
  </script>
  <div
    class="css-8f2oos"
  >
    <p
      class="css-uok8h6"
    >
      Other Categories
    </p>
    <div
      class="css-12atwa6"
    >
      <a
        class="chakra-link css-9dfuuy"
        href="/firstQuest/en"
      >
        1st Quest
      </a>
      <a
        class="chakra-link css-1ejw2rc"
        href="/secondQuest/en"
      >
        2nd Quest
      </a>
      <a
        class="chakra-link css-dbk65j"
        href="/swordless/en"
      >
        Swordless
      </a>
    </div>
  </div>
</DocumentFragment>
`;
